src/mpi_utils/Entangler.jl

```diff
@@ -172,13 +176,13 @@ function transmit!(e::Entangler, source_data::AbstractVector{T}, to_global_indic
             source_view = Ref{T}(source_datum)
             mpi_rank = process_index - 1
             # asynchronously (non-blocking) send over MPI:
-            Isend(source_view, e.communicator, dest = mpi_rank, tag = tag(e, transmit_index, global_index))
+            request = Isend(source_view, e.communicator, dest = mpi_rank, tag = tag(e, transmit_index, global_index))
+            push!(requests, request)
```

Comment:
Add a comment here stating that we wait for the Isend request to avoid the application terminating in the last iteration without completing its request.

Reply:
done!